src/core/electrostatics_magnetostatics/mmm2d.cpp

```diff
@@ -198,15 +198,17 @@ static double *gblcblk = nullptr;
 /** contribution from the image charges */
 static double lclimge[8];
 
-typedef struct {
+struct SCCache {
   double s, c;
-} SCCache;
+};
 
 /** sin/cos caching */
-static SCCache *scxcache = nullptr;
+static std::vector<SCCache> scxcache;
+/* _Not_ the size of scxcache */
 static int n_scxcache;
 /** sin/cos caching */
-static SCCache *scycache = nullptr;
+static std::vector<SCCache> scycache;
+/* _Not_ the size of scycache */
```

Comment:
If it's *not* the size of ```scycache``` could you please add a comment describing what it is?

Reply:
If only I would know. It's something like the logical size.